lib/appsignal/config.rb

```diff
@@ -538,6 +538,28 @@ def detect_from_system
         hash[:active] = true unless env_push_api_key.strip.empty?
 
         hash[:enable_at_exit_hook] = "always" if Appsignal::Extension.running_in_container?
+
+        # Set revision from REVISION file if present in project root
+        # This helps with Capistrano and Hatchbox.io deployments
+        revision_from_file = detect_revision_from_file
+        hash[:revision] = revision_from_file if revision_from_file
+      end
+    end
+
+    def detect_revision_from_file
+      logger.debug "Detecting revision from REVISION file in project root"
+      return unless root_path
+
+      revision_file_path = File.join(root_path, "REVISION")
+      logger.debug "Looking for REVISION file at: #{revision_file_path}"
+      return unless File.exist?(revision_file_path) && File.readable?(revision_file_path)
+
+      begin
+        revision_content = File.read(revision_file_path).strip
+        revision_content.empty? ? nil : revision_content
+      rescue StandardError => e
```

Comment:
```suggestion
      rescue SystemCallError => e
```

I think you want to rescue any kind of IO error? The [SystemCallError](https://ruby-doc.org/3.2.0/SystemCallError.html) class would be a better error class to rescue. StandardError is rescuing a lot more errors we should not expect to happen.

But on the other hand, you also already check if the file exists and is readable on line 555, so what different errors do you expect?